### PR TITLE
backport-2.0: ui: Auto-zoom map to fit visible content

### LINKED ISSUE #23513: ui: clusterviz: zoom to content
- On load of the node map, zoom to the minimum bounding box containing the child localities of the current locality.
- On clicking a locality, zoom to the minimum bounding box containing its sub-localities.

### PR DESCRIPTION
**Pull request by mrtracy**
The cluster map now auto-zooms to the set of visible information
whenever the user navigates to a new locality, or whenever new locality
information becomes available for the locality currently being viewed.

Fixes #23513

Release note: None

### CODE REVIEW

**Comment by vilterp** [comment]
LGTM, but let's hold off until we unbreak cluster viz by verifying, merging and cherry-picking #23879

**Review by vilterp — APPROVED** [review]
K, fix is merged, so we're good to go on this.

**Comment by couchand** [comment]
OTOH i raised some questions on the PR this cherry-picks that I think might be worth addressing.

**Comment by vilterp** [comment]
sure. up to @mrtracy whether to address them here or in his other zoom-related PR that hasn't been 
merged to master yet.

**Comment by mrtracy** [comment]
This cherry-pick now includes the follow-up commit #23883, which animates the transition.  This is ready for inclusion into 2.0 or 2.0.1, as deemed appropriate.

**Review by bdarnell — APPROVED** [review]

**Comment by bdarnell** [comment]
Let's merge this for 2.0; it'll be good for demos and is isolated from anything else.

**Pull request merged**
